docs/UsersGuide.adoc

```diff
@@ -516,6 +516,15 @@ Additional optional Options to configure the connection handling are:
 `:proxy-max-connection-retries`:: int, defaults to 1.
 `:proxy-max-request-time`:: ms as int, defaults to 30000. 30sec request timeout.
 
+Note: By default, with `:proxy-url` a request for `/` will 503, even if a `:root` contains a file named `index.html`. This behavior can be overriden link:++https://github.com/thheller/shadow-undertow/blob/07402ed93ee7c644dbdc761633fb1e48a27b5222/src/main/shadow/undertow.clj#L115++[through configuration] in https://github.com/thheller/shadow-undertow[shadow-undertow]. A config to support this might include:
```

Comment:
This statement isn't correct.

This will return whatever `https://some.host/` returns. If that answers with a `503` you get a `503`.

Sometimes you may not want to prefer loading index files over asking the proxy, which is what this option is for.

Reply:
I suppose this is somewhat of an unrelated issue that only shows because you are using `curl`, which I guess is a reasonable expectation that this would work.

Let me explain what is happening:

If you send a blank `curl localhost:8000` with no `:use-index-files true` set you indeed get a 404. This however isn't because the file doesn't exist, it is because the [push-state](https://github.com/thheller/shadow-cljs/blob/master/src/main/shadow/http/push_state.clj) handler that ends up getting called tries to be "smart". "push state" is weird name, but meant to enable "SPA" that want play with the browser url. With some kind of router you might set the path to `http://localhost:8000/foo`, if you then reload the page you'd get a 404. So push-state by default sends you the `index.html` contents instead. It however only does this after checking the `Accept` header usually send by the browser. This is so that image requets such as `/img/foo.png` still result in a 404 if not found, instead of also just getting sent the `index.html` contents, which make for a bad image. Same for missing JS files and so on. Basically this should only work for HTML requests, which is why the `Accept` header is checked.

So, the problem ends up being that browsers send this by default, but `curl` does not!

If you instead send `curl -H "Accept: text/html" http://localhost:8000` you indeed get the correct reply, with no need for `:use-index-files true`. This also makes the proxy work as expected as that header is transferred with the request.

This is of course kinda not mentioned anywhere, so I can understand the confusion. It probably should be.